torch/lib/THC/generic/THCTensorMathPointwise.cu

```diff
@@ -61,6 +60,13 @@ IMPLEMENT_CUDA_TENSOR_BASIC_FUNC( cinv, THCNumerics<real>::cinv,  Real)
 
 #endif
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF) || \
```

Comment:
I would still prefer `#if !defined(THC_REAL_IS_CHAR) && !defined(THC_REAL_IS_BYTE)` as its clearer to me. Up to you. Though I would say adding a comment about why `char` is not supported would be useful.